app/Http/Controllers/AuthController.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Rules\ComplexPassword;
+use Illuminate\Http\Request;
+use App\Models\User;
+use App\Mail\RegisterMail;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Mail;
+use Illuminate\Support\Str;
+
+class AuthController extends Controller
+{
+
+    public function create_user(Request $request)
+    {
+        request()->validate([
+            'username'=>'required|unique:users',
```

Comment:
I think is good idea to add |max:255|string|

Reply:
Done